File: ALGORITHMS/Matrix/Matrix Multiplication/matrixMultiplication.py

def multiply(m1, m2, mat1,
			n1, n2, mat2):

	res = [[0 for x in range(n2)]
			for y in range (m1)]
	
	for i in range(m1):
		for j in range(n2):
			res[i][j] = 0
			for x in range(m2):		
				res[i][j] += (mat1[ i][x] *
							mat2[ x][j])
			
	for i in range(m1):
		for j in range(n2):	
			print (res[i][j],
				end = " ")	
		print ()


if __name__ == "__main__":

	mat1 = [[2, 4], [3, 4]]
	mat2 = [[1, 2], [1, 3]]
	m1, m2, n1, n2 = 2, 2, 2, 2

	# Function call
	multiply(m1, m2, mat1,
			n1, n2, mat2)
	

File: ALGORITHMS/Matrix/Matrix Multiplication/matrixMultiplication.c
#include <stdio.h>


void multiply(int m1, int m2, int mat1[][m2], int n1,
			int n2, int mat2[][n2])
{
	int x, i, j;
	int res[m1][n2];
	for (i = 0; i < m1; i++)
	{
		for (j = 0; j < n2; j++)
		{
			res[i][j] = 0;
			for (x = 0; x < m2; x++)
			{
				*(*(res + i) + j) += *(*(mat1 + i) + x)
									* *(*(mat2 + x) + j);
			}
		}
	}
	for (i = 0; i < m1; i++)
	{
		for (j = 0; j < n2; j++)
		{
			printf("%d ", *(*(res + i) + j));
		}
		printf("\n");
	}
}


int main()
{
	int mat1[][2] = { { 2, 4 }, { 3, 4 } };
	int mat2[][2] = { { 1, 2 }, { 1, 3 } };
	int m1 = 2, m2 = 2, n1 = 2, n2 = 2;


	multiply(m1, m2, mat1, n1, n2, mat2);
	return 0;
}
